### PR TITLE
[Core] Replaced StopAsyncIteration with a return to prevent errors when no tasks are provided.

### PR DESCRIPTION
**Pull request by mk-armah**
# Description

**What**:
- Replaced `StopAsyncIteration` with a return statement in the `stream_async_iterators_tasks` function. This ensures nothing is returned when empty tasks are provided to the function.

**Why**:
- The use of `StopAsyncIteration` was causing improper error handling in scenarios where no async tasks were provided. By replacing with a return statement, the error is prevented and nothing is returned just as nothing was provided.

**How**:
- Replaced StopAsyncIteration with a return statement return nothing when an empty list of tasks is passed to the `stream_async_iterators_tasks` function.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Review by matan84 — CHANGES_REQUESTED** [review]
One comment

File: port_ocean/utils/async_iterators.py [hunk 1]
```diff
@@ -36,7 +37,7 @@ async def main():
     :return: A stream of results
     """
     if not tasks:
-        raise StopAsyncIteration("No tasks provided")
+        raise NoAsyncTasksFound("No or empty tasks provided")
```
[the remark this answers is not in the document]
Comment (mk-armah) [reply port_ocean/utils/async_iterators.py:40]: 🤔

[the remark this answers is not in the document]
Comment (mk-armah) [reply port_ocean/utils/async_iterators.py]: I think it's safer that way. 👍🏾

**Review by matan84 — APPROVED** [review]
LGTM

**Pull request merged**